srf_generation/poly_fault_select.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+"""
+Select faults based on being within a polygon.
+"""
+
+from argparse import ArgumentParser
+import os
+
+from matplotlib import pyplot as plt
+import numpy as np
+from osgeo import ogr, osr
+
+from qcore.nhm import load_nhm
+
+
+def points2polygon(points):
+    """
+    GDAL polygon from points (2D array).
+    https://pcjericks.github.io/py-gdalogr-cookbook/geometry.html
+    """
+    ring = ogr.Geometry(ogr.wkbLinearRing)
+    for ll in np.atleast_2d(points):
+        ring.AddPoint_2D(*map(float, ll))
+    ring.CloseRings()
+    poly = ogr.Geometry(ogr.wkbPolygon)
+    poly.AddGeometry(ring)
+    return poly
+
+
+def points2trace(points):
+    """
+    GDAL trace from points (2D array).
+    """
+    trace = ogr.Geometry(ogr.wkbLineString)
+    for ll in np.atleast_2d(points):
+        trace.AddPoint_2D(*map(float, ll))
+    return trace
+
+
+if __name__ == "__main__":
+    # arguments
+    parser = ArgumentParser()
+    parser.add_argument("nhm_file", type=os.path.abspath)
+    parser.add_argument("points", nargs="+", type=float)
```

Comment:
The only minor thing would be specifying what is needed from each argument in the help text. So a user can tell what format the points should be.